### PR TITLE
Externalising JS

### PR DESCRIPTION
**Pull request by gauntface**
This is a large number of commits, tried rebasing and squashing - failed miserably.

This ultimately externalises all JS files, apart from report-validity which breaks github pages (any directory beginning with "repo" will prevent include_relative from working.

### CODE REVIEW

**Comment by addyosmani** [comment]
High level comment: I think there's great value in extracting out the JS for all of our samples. It's easier to tinker and debug in DevTools (which doesn't have adequate support for inline-JS debugging), easier to run existing JS tooling on (linting, code-style w/o extraction) and generally (imo) makes the authoring experience a lot more pleasant.

:+1: from me.

**Comment by beaufortfrancois** [comment]
Woo! Thank you @gauntface!
Looks great to me.

File: styles/main.css [hunk 1]
```diff
@@ -24,6 +24,10 @@ h3 {
   margin-top: 2em;
 }
 
+.capitalise {
```
Comment (jeffposnick) [inline styles/main.css]: I don't see `capitalise` used as a class anywhere. Was this added inadvertently?

**Comment by jeffposnick** [comment]
Much appreciated, @gauntface! :+1:, other than that one comment.

There are a couple of in-flight samples that can be adjusted to the new format once this PR is merged.

**Comment by Psoffie009** [comment]
> This is a large number of commits, tried rebasing and squashing - failed miserably.
> 
> This ultimately externalises all JS files, apart from report-validity which breaks github pages (any directory beginning with "repo" will prevent include_relative from working.

**Comment by Psoffie009** [comment]
> javhd.com

**Comment by Psoffie009** [comment]
> Much appreciated, @gauntface! 👍, other than that one comment.
> 
> There are a couple of in-flight samples that can be adjusted to the new format once this PR is merged.

File: styles/main.css (hunk 1 above)
Comment (Psoffie009) [reply styles/main.css]: > I don't see `capitalise` used as a class anywhere. Was this added inadvertently?

**Comment by Psoffie009** [comment]
> Woo! Thank you @gauntface!
> Looks great to me.

**Comment by Psoffie009** [comment]
> This is a large number of commits, tried rebasing and squashing - failed miserably.
> 
> This ultimately externalises all JS files, apart from report-validity which breaks github pages (any directory beginning with "repo" will prevent include_relative from working.

**Comment by Psoffie009** [comment]
> High level comment: I think there's great value in extracting out the JS for all of our samples. It's easier to tinker and debug in DevTools (which doesn't have adequate support for inline-JS debugging), easier to run existing JS tooling on (linting, code-style w/o extraction) and generally (imo) makes the authoring experience a lot more pleasant.
> 
> 👍 from me.

**Pull request merged**